backend/controllers/User.js

```diff
@@ -1,19 +1,31 @@
-const User = require("../models/User");
-const { generateToken } = require("../utils/auth");
-const bcrypt = require("bcryptjs");
+const User = require('../models/User');
+const { generateToken } = require('../utils/auth');
+const bcrypt = require('bcryptjs');
 
-//Sign up function
+// Get leaderboard function
+const leaderboard = async (req, res) => {
+  try {
+    // get all users
+    const users = await User.find();
```

Comment:
If you want to get the top 10 users with scores then do something like this (be sure to verify if this is correct, e.g. change limit to 2 and confirm that it's indeed the 2 people with top score in your database, view it in your Mongo Atlas)
`User.find().sort({ score: -1 }).limit(10)`

Reply:
I tried this exact syntax but errors kept appearing ("users is undefinable variable") so we broke it down and only fetched users here and sorted them in frontend

Reply:
You can keep it as is for now, and let's do a quick sync over Discord/Zoom when we have time to fix this, since I do think it's better to deal with it on backend rather frontend